tests/unit/test_utils.py

```diff
@@ -157,3 +157,33 @@ def test_human_time_negative_float(data: HumanTimeTestData) -> None:
     """
     result = utils.human_time(-float(data.value))
     assert result == f"-{data.expected}"
+
+
+class RoundHalfUpTestData(NamedTuple):
+    """Data for round half up tests."""
+
+    id: str
+    value: Union[int, float]
+    expected: int
+
+
+round_half_up_test_data = [
+    RoundHalfUpTestData(id="integer", value=1, expected=1),
+    RoundHalfUpTestData(id="negative integer", value=-1, expected=-1),
+    RoundHalfUpTestData(id="down float", value=1.49999999, expected=1),
+    RoundHalfUpTestData(id="up float", value=1.50000001, expected=2),
+    RoundHalfUpTestData(id="negative down float", value=-1.49999999, expected=-1),
+    RoundHalfUpTestData(id="negative up float", value=-1.50000001, expected=-2),
+]
+
+
+@pytest.mark.parametrize("data", round_half_up_test_data, ids=lambda data: data.id)
+def test_round_half_up(data: RoundHalfUpTestData) -> None:
```

Comment:
To me, this looks like an ideal use case to integrate property-based testing: https://hypothesis.rtfd.io.

Reply:
that's a cool library, in this case we'd have to find a way to determine the expected result without using the function we were testing.  Certainly something to look at in the future.

Reply:
I've been looking for an excuse to integrate Hypothesis for years but never got to it :) I guess we need to start somewhere...